### PR TITLE
[Python] Enhancements

### PR DESCRIPTION
**Pull request by SirNickolas**
[PEP 3127](https://www.python.org/dev/peps/pep-3127/) proposed binary and octal prefixes for integers.

[Waf](https://waf.io/) is a build system which is written in Python and uses Python scripts to control the building process.

### CODE REVIEW

**Comment by wbond** [comment]
Thanks for contributing!

Could you add some tests for the integer forms?

**Comment by buildersbrewery** [comment]
@SirNickolas http://www.sublimetext.com/docs/3/syntax.html#Testing

File: Python/Python.sublime-syntax [hunk 1]
```diff
@@ -14,6 +14,7 @@ file_extensions:
   - gyp
   - gypi
   - Snakefile
+  - wscript
```
Comment (keith-hall) [inline Python/Python.sublime-syntax]: my understanding from reading https://waf.io/book/#_using_several_scripts_and_folders is that `wscript` is not a file extension, but the full filename of a Waf project description file.

Comment (keith-hall) [inline Python/Python.sublime-syntax]: and these files should be picked up as Python by the `first_line_match` anyway: https://waf.io/book/#_declaring_waf_commands

**Comment by SirNickolas** [comment]
> `wscript` is not a file extension, but the full filename of a Waf project description file.

Indeed it is.

> and these files should be picked up as Python by the `first_line_match` anyway

`wscript`s are not executables, and they are not invoked directly, but via `waf` command. It compiles and `exec`s their code. So the first line may actually be any valid Python code.

**Comment by keith-hall** [comment]
>> wscript is not a file extension, but the full filename of a Waf project description file.

> Indeed it is.

so it should be removed from the syntax definition's file extensions then, as it won't work

> the first line may actually be any valid Python code.

maybe there is no reliable way to do it without using a plugin to check the full filename

**Comment by wbond** [comment]
> so it should be removed from the syntax definition's file extensions then, as it won't work

No, the Ruby syntax supports an extensive number of extension-less filenames. I believe the filename checker checks the whole filename if there is no `.` in it.

**Comment by keith-hall** [comment]
> I believe the filename checker checks the whole filename if there is no . in it.

oh I didn't know that. But it does mean that it will incorrectly treat files with a wscript extension as python...

**Comment by SirNickolas** [comment]
> as it won't work

Well, but it does work. Look, just a line above there is `Snakefile` definition, which is exactly the same - a file without an extension, just named "Snakefile".

> it will incorrectly treat files with a wscript extension as python

Who and for what reason will assign a `.wscript` extension to a file which is not a `wscript`?

**Comment by wbond** [comment]
I just looked online and found no reference to the `.wscript` extension. There is `WScript.exe`, but it uses `.wsf`, `.vbs` or `.js`.

**Comment by keith-hall** [comment]
In which case, I retract my comments :) sorry for causing a fuss!

**Comment by wbond** [comment]
@keith-hall No worries, I (and I'm sure many others in the community) appreciate all of the time you take double-checking things!

**Comment by wbond** [comment]
Excellent, thanks for this @SirNickolas!

**Pull request merged**